TWTValidation/JSON Validator/Processors/TWTJSONSchemaPrettyPrinter.m

```diff
@@ -0,0 +1,279 @@
+//
+//  TWTJSONSchemaPrettyPrinter.m
+//  TWTValidation
+//
+//  Created by Jill Cohen on 1/7/15.
+//  Copyright (c) 2015 Two Toasters, LLC.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "TWTJSONSchemaPrettyPrinter.h"
+
+#import "TWTJSONSchemaASTCommon.h"
+#import "TWTJSONSchemaKeywordConstants.h"
+
+@interface TWTJSONSchemaPrettyPrinter ()
+
+@property (nonatomic, copy) NSDictionary *topLevelSchema;
```

Comment:
I want to emphasize the fact that the top level structure in a JSON schema is always a dictionary. (It can't be an array of schemas, for example).
Also, I've been using "schema" to mean an NSDictionary that's a JSON schema and "node" to mean TWTJSONSchemaASTNode, which could represent a whole or a part of a JSON schema. It's consistent with the parser file in this respect.